hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/LocalQueueStatsImpl.java

```diff
@@ -59,7 +59,7 @@
     @Probe(name = QUEUE_METRIC_BACKUP_ITEM_COUNT)
     private int backupItemCount;
     @Probe(name = QUEUE_METRIC_MIN_AGE, unit = MS)
-    private long minAge;
+    private long minAge = Long.MAX_VALUE;
```

Comment:
What is the reasoning behind this change? I have read the discussion with Emre, but couldn't understand why we only change `minAge`. What about `maxAge` just below?

Reply:
Hello Metin,

in short, we wanted to make the behavior consistent in a sense that the `minAge` metric has the same value initially and after reset.
The problem was, this didn't work out of the box, because there are two defaullt values used here:
 - initially, the default of `LocalMapStatsImpl#minAge` is used, because the `QueueService` returns an empty LocalMapStatsImpl if there
 is no QueueContainer created yet (see [here](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueService.java#L344)).
 - after reset, when there is an existing `QueueContainer` but there were no interaction with it since the reset, then the `LocalMapStatsImpl#minAge` will be set by
 `QueueContainer#setStats()` , so in the end the `minAge` value set by the `#resetAgeStats()` will be returned as the metric value to the client. 

So we need to set these two initial values to the same value (in `LocalMapStatsImpl#minAge` and `QueueContainer#resetAgeStats()`) so that the behavior is consistent.
If we would use `0` at both places (so that we don't have to touch the `LocalMapStatsImpl#minAge` default), then we would need to assign `minAge = 0` in the `#resetAgeStats()` as well, but that would break the metric: in that case the `minAge` would always remain `0`: the [`Math.min()` call in the `#age()` method](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java#L1111) would always evaluate to `0`.

So we had to take the other path, and set `minAge` to `Long.MAX_VALUE` both in the `#resetAgeStats()` method and the `LocalMapStatsImpl#minAge` declaration.



The `maxAge` is not affected by the same problem, if we set it to `0` after the reset, then the `Math.max()` in the `age()` method will properly increase it as expected, and `LocalMapStatsImpl#maxAge` is also `0`.


I hope this answers your question.

Reply:
Thanks for the explanation. I still feel like it would be best to have just one default value for `maxAge` as well (`Long.MIN_VALUE`).
In a totally imaginary scenario below, the user would see two different values before & after the reset for `maxAge`.

```
        HazelcastInstance instance = Hazelcast.newHazelcastInstance();
        IQueue<Object> queue = instance.getQueue("test");
        System.out.println(queue.getLocalQueueStats().getMaxAge()); // 0
        queue.isEmpty();
        System.out.println(queue.getLocalQueueStats().getMaxAge()); // Long.MIN_VALUE
        // reset the stats
        System.out.println(queue.getLocalQueueStats().getMaxAge()); // 0
```

Reply:
Yes, why not. I updated the PR and extracted the default values into constants so if later we decide to change them, enough to change at one place.